### PR TITLE
Improved logging

### PR DESCRIPTION
**Pull request by elandau**
- Improve logging by linking cause to WebApplicationException
- Add attempt counter to TimeSupervisorTask to help identify situations where the supervisor has stalled

### CODE REVIEW

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/jersey/AbstractJerseyEurekaHttpClient.java [hunk 1]
```diff
@@ -201,6 +201,14 @@ protected AbstractJerseyEurekaHttpClient(Client jerseyClient, String serviceUrl)
                     .headers(headersOf(response))
                     .entity(applications)
                     .build();
+        } catch (Exception e) {
+            logger.info("Error invoking Jersey HTTP GET {}/{}?{}; statusCode={}",
+                    serviceUrl, urlPath,
+                    regionsParamValue == null ? "" : "regions=" + regionsParamValue,
+                    response == null ? "N/A" : response.getStatus(),
+                    e
```
[the remark this answers is not in the document]
Comment (elandau) [reply eureka-client/src/main/java/com/netflix/discovery/shared/transport/jersey/AbstractJerseyEurekaHttpClient.java]: This code is fairly deep down the call stack and I don't want to risk affecting any code that might already depend on specific exceptions by wrapping this.  What I really want here is to be able to log the service URL and any additional context that would help us debug these exceptions further.

File: eureka-client/src/main/java/com/netflix/discovery/TimedSupervisorTask.java [hunk 2]
```diff
@@ -59,6 +61,7 @@ public TimedSupervisorTask(String name, ScheduledExecutorService scheduler, Thre
     public void run() {
         Future<?> future = null;
         try {
+            attemptCounter.increment();
```
[the remark this answers is not in the document]
Comment (elandau) [reply eureka-client/src/main/java/com/netflix/discovery/TimedSupervisorTask.java]: sure

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/jersey/AbstractJerseyEurekaHttpClient.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (qiangdavidliu) [reply eureka-client/src/main/java/com/netflix/discovery/shared/transport/jersey/AbstractJerseyEurekaHttpClient.java]: 👍 
I poked through the code to check what actually happens to the exception (before it was implicitly rethrown, and now it is explicitly rethrown). I see that the full exception is eventually logged at:
https://github.com/Netflix/eureka/blob/master/eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java#L817
https://github.com/Netflix/eureka/blob/master/eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java#L973
so if you want to log the full exception here, may be remove the duplicate logging in these other places?

[the remark this answers is not in the document]
Comment (elandau) [reply eureka-client/src/main/java/com/netflix/discovery/shared/transport/jersey/AbstractJerseyEurekaHttpClient.java]: I removed this logging as I actually don't think it'll help us much anyway.  The other changes provide more meaningful information.

**Comment by qiangdavidliu** [comment]
ship it!

**Pull request merged**